src/engine/sparqlExpressions/SparqlExpressionGenerators.h

```diff
@@ -71,26 +87,53 @@ CPP_template(typename T, typename Transformation = std::identity)(
 }
 
 template <typename Transformation = std::identity>
-inline cppcoro::generator<
-    const std::decay_t<std::invoke_result_t<Transformation, Id>>>
+inline ad_utility::InputRangeTypeErased<
+    std::decay_t<std::invoke_result_t<Transformation, Id>>>
 resultGenerator(ad_utility::SetOfIntervals set, size_t targetSize,
                 Transformation transformation = {}) {
-  size_t i = 0;
-  const auto trueTransformed = transformation(Id::makeFromBool(true));
-  const auto falseTransformed = transformation(Id::makeFromBool(false));
-  for (const auto& [begin, end] : set._intervals) {
-    while (i < begin) {
-      co_yield falseTransformed;
```

Comment:
Same as below, this is something like `set | views::transform(return repeatView) | views::join`
But I'll have to look at the concrete implementations.